packages/backend/src/apps/gathersg/dynamic-data/get-case-fields.ts

```diff
@@ -5,54 +5,107 @@ import {
 } from '@plumber/types'
 
 import HttpError from '@/errors/http'
+import { computeForEachParameters } from '@/helpers/compute-for-each-parameters'
+import computeParameters, { variableRegExp } from '@/helpers/compute-parameters'
+import { getTestExecutionSteps } from '@/helpers/get-test-execution-steps'
 
-import { fetchCaseFields } from '../common/fetch-case-fields'
-import { GatherSGError } from '../common/types'
-
-/**
- * Subset of result
- */
-interface GatherSGCase {
-  uuid: string
-  createdAt: string
-  updatedAt: string
-  status: {
-    uuid: string
-    name: string
-    color: string
-    isFinal: boolean
+import { fetchCaseFields, GatherSGCaseField } from '../common/fetch-case-fields'
+import { GatherSGCase, GatherSGError } from '../common/types'
+
+const getCaseUuidFromVariable = async (
+  $: IGlobalVariable,
+  variable: string,
+) => {
+  let computedCaseUuid
+  const testExecutionSteps = await getTestExecutionSteps($.flow.id)
+
+  if (/items.columns/.test(variable)) {
+    // if variable from for each, then we compute slightly differently
+    const stepIdAndKeyPath = variable.replace(/{{step.|}}/g, '') as string
+    const [stepId, ...keyPaths] = stepIdAndKeyPath.split('.')
+
+    const executionStep = testExecutionSteps.find((executionStep) => {
+      return executionStep.stepId === stepId
+    })
+
+    computedCaseUuid = computeForEachParameters({
+      data: executionStep?.dataOut,
+      keyPath: keyPaths.join('.'),
+      executionSteps: testExecutionSteps,
+      executionStep,
+      stepId,
+      forEachContext: {
+        executionStepMetadata: testExecutionSteps[0].metadata,
+        forEachStepPosition: 0,
+        isForEachStep: true,
+        stepPositions: {
+          [stepId]: 0,
+        },
+      },
+    })
+  } else {
+    const { caseUuid } = computeParameters(
+      { caseUuid: variable },
+      testExecutionSteps,
+    )
+    computedCaseUuid = caseUuid
+  }
+
+  return computedCaseUuid ?? '`'
+}
+
+const processCaseFields = (caseFields: GatherSGCaseField[]) => {
+  return {
+    data: caseFields.map((field) => {
+      return {
+        name: field.name,
+        value: field.name,
+      }
+    }),
   }
-  caseRef: string
-  fields: Record<string, string | string[] | null | number>
-  tags: string[]
 }
 
 const dynamicData: IDynamicData = {
   key: 'getCaseFields',
   name: 'Get Case Fields',
   async run($: IGlobalVariable): Promise<DynamicDataOutput> {
     try {
-      const { caseType: caseTypeUuid } = $.step.parameters
+      const { caseType: caseTypeUuid, caseUuid } = $.step.parameters
 
       if (caseTypeUuid) {
+        // Fetch case fields using the determined case type UUID
         const { filteredFields } = await fetchCaseFields({
           $,
           caseTypeUuid: caseTypeUuid as string,
         })
 
+        return processCaseFields(filteredFields)
+      } else if (caseUuid) {
+        if (typeof caseUuid === 'string' && caseUuid.match(variableRegExp)) {
```

Comment:
suggestion: if u wanna match the string entirely
```suggestion
        if (typeof caseUuid === 'string' && caseUuid.match(`^${VARIABLE_REGEX.source}$`) {
```